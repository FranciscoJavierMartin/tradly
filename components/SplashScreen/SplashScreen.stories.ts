import SplashScreen from './SplashScreen.vue';

export default {
  title: 'SplashScreen',
  component: SplashScreen,
  args: {},
  parameters: {
    layout: 'fullscreen',
  },
};

const Template = (args) => ({
  components: { SplashScreen },
  setup() {
    return { args };
  },
  template: '<SplashScreen v-bind="args"/>',
});

export const Basic = Template.bind({});
Basic.args = {};
