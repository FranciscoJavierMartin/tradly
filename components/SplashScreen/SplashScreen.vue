<template>
  <div class="c-splash-screen">
    <img src="../../assets/imgs/logo.svg" />
    <h2>Tradly</h2>
  </div>
</template>

<style lang="scss">
@import '../../assets/styles/main.scss';

.c-splash-screen {
  background-color: $dark-green;
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  h2 {
    color: $white;
    font-weight: 500;
    font-size: 30px;
    line-height: 30px;
    text-align: center;
  }
}
</style>
