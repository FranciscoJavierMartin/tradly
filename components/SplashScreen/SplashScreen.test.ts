import '@testing-library/jest-dom';
import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/vue';
import SplashScreen from './SplashScreen.vue';

describe('SplashScreen.vue', () => {
  test('should render', () => {
    const { container } = render(SplashScreen);

    const heading = screen.getByRole('heading', { name: 'Tradly' });
    expect(heading).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
