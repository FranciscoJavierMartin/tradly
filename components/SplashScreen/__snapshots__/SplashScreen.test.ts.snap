// Vitest Snapshot v1

exports[`SplashScreen.vue > should render 1`] = `
<div
  class="c-splash-screen"
>
  <img
    src="/assets/imgs/logo.svg"
  />
  <h2>
    Tradly
  </h2>
</div>
`;
